### PR TITLE
Ovirt add rename functionality

### LINKED ISSUE #43236: ovirt module: Rename ovirt management network
##### SUMMARY

During the deployment on ovirt Hosted-Engine via ansible I need to rename the network from the default name (ovirtmgmt) to something else. Currently there is no way to do it on the ovirt-ansible modules.

This RFE is crucial for Hosted-engine to support other management network names other than the default one.

##### ISSUE TYPE
 - Feature idea

##### COMPONENT NAME
ovirt-ansible module

##### ANSIBLE VERSION
2.6

### PR DESCRIPTION
**Pull request by mnecas**
##### SUMMARY

Now you can rename most of ovirt components. When you pass components ID then you can change name.  

Fixes #43236

##### ISSUE TYPE

 - Feature Pull Request
 - Bugfix Pull Request
##### COMPONENT NAME

ovirt
##### ANSIBLE VERSION

```
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/mnecas/Desktop/Projects/Redhat/ansible/lib/ansible/modules/cloud/ovirt']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Review by mwperina — APPROVED** [review]
shipit

File: lib/ansible/modules/cloud/ovirt/ovirt_cluster.py [hunk 1]
```diff
@@ -266,6 +270,11 @@
 - ovirt_cluster:
     state: absent
     name: mycluster
+
+# Change cluster Name
+- ovirt_cluster:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new cluster name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_cluster.py]: There can't be spaces in cluster name.

File: lib/ansible/modules/cloud/ovirt/ovirt_datacenter.py [hunk 2]
```diff
@@ -77,6 +81,11 @@
 - ovirt_datacenter:
     state: absent
     name: mydatacenter
+
+# Change Datacenter Name
+- ovirt_datacenter:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new datacenter name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_datacenter.py]: Here as well.

File: lib/ansible/modules/cloud/ovirt/ovirt_host.py [hunk 3]
```diff
@@ -18,6 +18,10 @@
 description:
     - "Module to manage hosts in oVirt/RHV"
 options:
+    id:
+        description:
+            - "ID of the nic to manage."
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host.py]: s/id/host

File: lib/ansible/modules/cloud/ovirt/ovirt_mac_pool.py [hunk 4]
```diff
@@ -58,6 +62,11 @@
 - ovirt_mac_pool:
     state: absent
     name: mymacpool
+
+# Change MAC pool Name
+- ovirt_nic:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new mac pool name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_mac_pool.py]: No spaces.

File: lib/ansible/modules/cloud/ovirt/ovirt_network.py [hunk 5]
```diff
@@ -93,6 +97,12 @@
 - ovirt_network:
     state: absent
     name: mynetwork
+
+# Change Network Name
+- ovirt_network:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new network name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_network.py]: No spaces.

File: lib/ansible/modules/cloud/ovirt/ovirt_nic.py [hunk 6]
```diff
@@ -94,6 +98,12 @@
     state: absent
     vm: myvm
     name: mynic
+
+# Change NIC Name
+- ovirt_nic:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new nic name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_nic.py]: No spaces.

File: lib/ansible/modules/cloud/ovirt/ovirt_quota.py [hunk 7]
```diff
@@ -106,6 +110,12 @@
     state: absent
     data_center: dcX
     name: quota1
+
+# Change Quota Name
+- ovirt_quota:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new quota name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_quota.py]: No spaces.

File: lib/ansible/modules/cloud/ovirt/ovirt_tag.py [hunk 8]
```diff
@@ -93,6 +97,11 @@
 - ovirt_tag:
     state: absent
     name: mytag
+
+# Change cluster Name
+- ovirt_tag:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new tag name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_tag.py]: No spaces. s/new tag name/new_cluster_name

File: lib/ansible/modules/cloud/ovirt/ovirt_template.py [hunk 9]
```diff
@@ -185,6 +185,11 @@
     state: absent
     name: mytemplate
 
+# Change Template Name
+- ovirt_template:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new template name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_template.py]: No spaces.

File: lib/ansible/modules/cloud/ovirt/ovirt_vm.py [hunk 10]
```diff
@@ -677,6 +677,11 @@
     nics:
       - name: nic1
 
+# Change VM Name
+- ovirt_vmpool:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new vm name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vm.py]: No spaces.

File: lib/ansible/modules/cloud/ovirt/ovirt_vm.py [hunk 11]
```diff
@@ -677,6 +677,11 @@
     nics:
       - name: nic1
 
+# Change VM Name
+- ovirt_vmpool:
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vm.py]: `ovirt_vm`

File: lib/ansible/modules/cloud/ovirt/ovirt_vmpool.py [hunk 12]
```diff
@@ -97,6 +101,11 @@
 - ovirt_vmpool:
     state: absent
     name: myvmpool
+
+# Change Pool Name
+- ovirt_vmpool:
+    id: 00000000-0000-0000-0000-000000000000
+    name: "new disk name"
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vmpool.py]: No spaces. s/new disk name/new_pool_name

File: lib/ansible/modules/cloud/ovirt/ovirt_host.py (hunk 3 above)
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host.py]: s/nic/host

**Comment by machacekondra** [comment]
shipit

**Review by mwperina — APPROVED** [review]
shipit

**Comment by mwperina** [comment]
Ondro, maybe you will need to approve the PR to get this merged automatically?
@machacekondra

**Comment by mwperina** [comment]
bot_status

**Review by machacekondra — APPROVED** [review]
shipit

**Comment by mwperina** [comment]
bot_status

**Comment by machacekondra** [comment]
shipit

**Comment by mwperina** [comment]
We have found out on Thursday that there was some misinformation about Community Freeze dates (already discussed that with Dylan). Ryan, is it possible to get this to 2.7? It's partial bug and partial RFE (only becuase of the need to add ID parameter) and it's quite critical for RHV.

@ryansb

**Comment by mwperina** [comment]
+label bug

**Comment by mwperina** [comment]
+label bug

**Comment by sandrobonazzola** [comment]
@ryansb any update?

**Comment by abadger** [comment]
I'm sorry, this feature wasn't even submitted until Friday, the day after final feature freeze. for 2.7.  We can look at getting this into the 2.8 release but it is too late to make 2.7.

**Comment by abadger** [comment]
Other than version_added being updated to say "2.8" instead of 2.7, I think this is okay.  Once version_added has been updated, this can be merged into devel.

**Review by mwperina — APPROVED** [review]
shipit

**Comment by abadger** [comment]
merged to devel for the 2.8 release.

**Comment by jinh-dk** [comment]
Can I rename oVirt template with this PR?

**Comment by machacekondra** [comment]
> Can I rename oVirt template with this PR?

Yes, if you pass id of the old template in `id`, and new name in `name` parameter, the template will be renamed.

**Pull request merged**